code/midtown/mmdyna/bndtmpl.cpp

```diff
@@ -20,6 +20,74 @@ define_dummy_symbol(mmdyna_bndtmpl);
 
 #include "bndtmpl.h"
 
+#include "vector7/geomath.h"
+
+#include "bndtmpl2.h"
+#include "poly.h"
+
+void mmBoundTemplate::ComputeBounds()
+{
+    GetBoundInfo(NumVerts, Verts, &BBMin, &BBMax, &Center, &Radius);
+    RadiusSqr = Radius * Radius;
+}
+
+void mmBoundTemplate::ComputeEdges()
+{
+    i32 total_verts = 0;
+
+    for (i32 i = 0; i < NumPolys; ++i)
+    {
+        total_verts += Polygons[i].GetNumVerts();
+    }
+
+    i32* edge_1s = new i32[4 * total_verts];
+    i32* edge_2s = new i32[4 * total_verts];
+
+    i32 num_edges = 0;
+
+    for (i32 j = 0; j < NumPolys; ++j)
+    {
+        mmPolygon& poly = Polygons[j + 1];
+
+        i32 num_verts = poly.GetNumVerts();
+
+        i32 v1 = poly.MtlIndex + num_verts; // ?
+        // Original line: v1 = *((__int16 *)&this->Polygons[j + 1].MtlIndex + num_verts);
+
+        for (i32 k = 0; k < num_verts; ++k)
+        {
+            i32 v2 = poly.VertIndices[k];
+
+            if (!EdgeInList(v1, v2, num_edges, edge_1s, edge_2s))
+            {
+                edge_1s[num_edges] = v1;
+                edge_2s[num_edges++] = v2;
```

Comment:
I'd rather move the `num_edges++` to a separate line after, since multiple things rely on it in this branch.